proposed/2021/TargetFrameworkAndPlatformsFilters.md

```diff
@@ -0,0 +1,196 @@
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/SupportedPlatforms.png)
+
+#### Version Numbers
+
+There are many unique version numbers for all generations of .NET & supported platforms. The filtering experience should be smart enough to create a version list based on the current selection. Here are a few scenarios:
+
+- User selects `Any` for the .NET generation.
+  - Version filter populates all known versions for .NET Framework, .NET Core, .NET, and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework, .NET Core, .NET, and .NET Standard.
+  - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8, 5.0, 6.0
+- User selects `Any` for the .NET generation and `Android` for the supported platform.
+  - Version filter populates all known version for .NET and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 6.0
+- User selects `.NET Framework` for the .NET generation.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Framework` for the .NET generation and `Windows` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 3.5, 4.0, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1, 4.7.2, 4.8
+- User selects `.NET Core` for the .NET generation and `macOS` for the supported platform.
+  - Version filter populates all known version for .NET Framework and .NET Standard.
+    - 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 2.0, 2.1, 2.2, 3.0, 3.1
+
+For further scenarios, see the [framework precedence](https://nugettools.azurewebsites.net/6.0.0-preview.3/framework-precedence) tool to understand how the version matrix would be calculated.
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/VersionNumbers.png)
+
+#### Complete Experience
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/CompleteExperience.png)
+
+## Search Syntax
+
+Lastly, NuGet should add a protocol in which supports [search syntax](https://docs.microsoft.com/nuget/consume-packages/finding-and-choosing-packages#search-syntax) for frameworks, platforms, and potentially *implementations**.
+
+Here's an ideal `<property>:<term>` for each use-case:
+
+```
+framework:net472 framework:net471 framework:net47   # Match the target framework monikers
```

Comment:
What would be the behavior when a user selects several platforms/frameworks in the UX, will it be 'AND' or 'OR' between the selections? (this might be out of scope for this spec)

Reply:
I believe current search syntax only supports `OR`. The examples here match what is currently supported.

Supporting `AND` and other advanced search syntax qualifiers would be nice & is likely supported:

https://docs.microsoft.com/en-us/azure/search/query-simple-syntax

For example, the `+` operator on a NuGet Azure Search query seems to work and support `AND`.

Reply:
> I believe current user-facing search syntax only supports `OR`.

We `OR` terms for the same field but you must match at least one term per field. For example imagine my query is `color:blue color:red make:Toyota make:Mazda`: this would match “blue Toyota”, or “blue Mazda”, but not “blue Hummer”.

> Supporting `AND` and other advanced search syntax qualifiers would be nice & is likely supported

Yup this could be supported, but we should be thoughtful here as the search syntax is quite confusing. Making the same syntax have different behaviors on different fields may complicate things further. 

Alternatively, we could instead revamp the current search syntax to be more sensible. We would need to understand how the current search syntax is used to make sure such a revamp is successful.

Reply:
Resolving as this has a path forward.